python/usepyhf/inference.py

```diff
@@ -0,0 +1,164 @@
+    print(f"Calculating p-values for POI {poi_name} = 0:")
+
+    #Calculate and print p-values
+    results = pyhf.infer.hypotest(
+            0, data, model, test_stat="q0", return_tail_probs=True,
+            fixed_params=fix_pars
+        )
+    print("")
+    print(f"CLb: {results[0]:.6f}")
+    print(f"CLs+b: {results[1][0]:.6f}")
+    print(f"CLs: {(results[1][0]/results[0]):.6f}")
+    print("")
+
+    print(f"Null hypothesis p-value: {results[0]:.6f}")
+    print(f"Significance level: {pValueToSignificance(results[0], True):.6f}")
+    print(f"Significance level needed to claim discovery: Z > 5")
+    print("")
+
+
+def p_values_excl(workspace:pyhf.Workspace):
+    """
+    Hypothesis test for the exclusion of a signal model.
+    Null hypothesis: Signal + background model
+    If the null hypothesis can be rejected at 1.65 significance
+    (p-value < 0.05) then the signal model can be excluded.
+    Uses the qtilde test statistics and sets POI to 1.
+    If the model has the SigXsec parameter, run test at SigXsec = [0, 1, -1]
+    """
+
+    #Extract model and data
+    model = workspace.model()
+    data = workspace.data(model)
+
+    #Find parameter of interest
+    poi_index = model.config.poi_index
+    poi_name = model.config.par_order[poi_index]
+
+    #Find SigXSec
+    par_order = model.config.par_order
+    init_pars = model.config.suggested_init()
+
+    #Print info
+    print("<INFO> pyhf: Running an exclusion hypothesis test:")
+
+    if "SigXsec" in par_order:
+        #Fix the parameter "SigXsec" if it exists
+        fix_pars = util.fix_sigxsec(model)
+        sigXsec_idx = [i for i, x in enumerate(fix_pars) if x][0] #sigXsec is only True in list
+        sigXSec_values = [0, 1, -1]
+        CLs=[]
+        CLb=[]
+        CLsplusb=[]
+        #Iterate over sigXSec=[0, 1, -1]
+        print(f"Suggested bounds for POI: { model.config.suggested_bounds()[poi_index] }")
+        print(f"Calculating p-values for POI {poi_name} = 1 and sigXSec = {sigXSec_values}:")
+        print("")
+        for value in sigXSec_values:
+            temp_inits = init_pars.copy()
+            temp_inits[sigXsec_idx] = value
+            #Calculate and print p-values
+            results = pyhf.infer.hypotest(
+                    1, data, model, test_stat="qtilde", return_tail_probs=True,
+                    init_pars = temp_inits, fixed_params=fix_pars
+                )
+            CLs.append(results[0])
+            CLb.append(results[1][1])
+            CLsplusb.append(results[1][0])
+        
+        print(f"CLs: [{CLs[0]:.6f}, {CLs[1]:.6f}, {CLs[2]:.6f}]")
+        print(f"CLb: [{CLb[0]:.6f}, {CLb[1]:.6f}, {CLb[2]:.6f}]")
+        print(f"CLs+b: [{CLsplusb[0]:.6f}, {CLsplusb[1]:.6f}, {CLsplusb[2]:.6f}]")
+        print("")
+
+        print(f"Null hypothesis p-value: {CLs[0]:.6f}")
+        print(f"Significance level: {pValueToSignificance(CLs[0], True):.6f}")
+        print(f"Significance level needed for exclusion: Z > 1.64")
```

Comment:
Please also remove these two lines reporting a significance since we just want the CLs.

Reply:
I don't understand why we report significance in the discovery test and not in the exclusion test?

Reply:
It's mostly following norms.  For a discovery we do care about the p-value and because we are familiar with levels of significance, and have chosen specific levels of 3 and 5 as important, use the significance to quantify deviations.  For exclusion tests, we chose to perform these at 95% CL, thus the threshold is at CLs=0.05, not the equivalent significance.